tests/common/rpc_api.py

```diff
@@ -1,21 +1,65 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
 from __future__ import annotations
-
-from base64 import b64encode
+from typing import AsyncContextManager, Dict, Optional, Callable, Any
+import trio
+from base64 import b64decode, b64encode
 import pytest
-from typing import Dict, Optional, Callable
-from quart.typing import TestClientProtocol
+import msgpack
+from quart import Quart
+from quart.typing import TestClientProtocol, TestHTTPConnectionProtocol
+from contextlib import asynccontextmanager
 from werkzeug.datastructures import Headers
+from dataclasses import dataclass
 
-from parsec._parsec import DateTime, OrganizationID
+from parsec._parsec import DateTime, OrganizationID, EventsListenRep
 from parsec.api.version import API_VERSION
 from parsec.core.types import LocalDevice
-from parsec.api.protocol import authenticated_ping_serializer, invited_ping_serializer
+from parsec.api.protocol import (
+    authenticated_ping_serializer,
+    invited_ping_serializer,
+)
+
+from tests.common import OrganizationFullData
+
 
-from tests.common import RunningBackend, OrganizationFullData
+class BaseRpcApiClient:
+    async def send(
+        self,
+        req,
+        serializer,
+        extra_headers: Dict[str, str] = {},
+        before_send_hook: Optional[Callable] = None,
+        now: Optional[DateTime] = None,
+        check_rep: bool = True,
+    ):
+        raise NotImplementedError
+
+
+@dataclass
+class SSEEventSink:
+    connection: TestHTTPConnectionProtocol
+    _status_code: int | None = None
+
+    @property
+    def status_code(self) -> int:
+        # Status code only available after the first event has been received
+        assert self.connection.status_code is not None
+        return self.connection.status_code
+
+    async def get_next_event(self, raw: bool = False) -> bytes:
+        data = await self.connection.receive()
+        if self._status_code is None:
+            self._status_code = self.connection.status_code
+        assert data.startswith(b"data:")
+        assert data.endswith(b"\n\n")
+        raw_event = b64decode(data[len("data:") : -2])
```

Comment:
Apparently, a space after `:` should be ignored:
> The following stream fires two identical events:
  ```
  data:test

  data: test
  ```
>  This is because the space after the colon is ignored if present.

https://html.spec.whatwg.org/multipage/server-sent-events.html

Reply:
done.